cli/cmd/cluster.go

```diff
@@ -0,0 +1,208 @@
+package cmd
+
+import (
+	"fmt"
+	"strings"
+
+	v1 "k8s.io/api/rbac/v1"
+
+	"github.com/linkerd/linkerd2/pkg/k8s"
+	"github.com/spf13/cobra"
+	corev1 "k8s.io/api/core/v1"
+	rbacv1 "k8s.io/api/rbac/v1"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/yaml"
+
+	"k8s.io/client-go/tools/clientcmd"
+	"k8s.io/client-go/tools/clientcmd/api"
+)
+
+const tokenKey = "token"
+
+type getCredentialsOptions struct {
+	namespace      string
+	serviceAccount string
+	clusterName    string
+}
+
+type createOptions struct {
+	serviceAccount string
+}
+
+func newCmdCluster() *cobra.Command {
+
+	getOpts := getCredentialsOptions{}
+	createOpts := createOptions{}
+
+	clusterCmd := &cobra.Command{
+		Use:   "cluster manages the multicluster  setup for Linkerd",
+		Short: "cluster manages the multicluster  setup for Linkerd",
+		Args:  cobra.NoArgs,
+	}
+
+	createCredentalsCommand := &cobra.Command{
+		Use: "create-credentials", Short: "Create the necessary credentialsfor service mirroring",
+
+		Args: cobra.NoArgs,
+		RunE: func(cmd *cobra.Command, args []string) error {
+
+			labels := map[string]string{
+				k8s.ControllerComponentLabel: k8s.ServiceMirrorLabel,
+				k8s.ControllerNSLabel:        controlPlaneNamespace,
+			}
+
+			clusterRole := v1.ClusterRole{
+				ObjectMeta: metav1.ObjectMeta{Name: createOpts.serviceAccount, Namespace: controlPlaneNamespace, Labels: labels},
+				TypeMeta:   metav1.TypeMeta{Kind: "ClusterRole", APIVersion: "rbac.authorization.k8s.io/v1"},
+				Rules: []rbacv1.PolicyRule{
+					{
+						APIGroups: []string{""},
+						Resources: []string{"services"},
+						Verbs:     []string{"list", "get", "watch"},
+					},
+				},
+			}
+
+			svcAccount := corev1.ServiceAccount{
+				ObjectMeta: metav1.ObjectMeta{Name: createOpts.serviceAccount, Namespace: controlPlaneNamespace, Labels: labels},
```

Comment:
I think `create-credentials` should also expose a `--namespace` flag as `get-credentials` does, instead of defaulting to `controlPlaneNamespace` which might not be relevant in a (possible not meshed) remote cluster. Also WDYT about having `--namespace` _not_ defaulting to "linkerd", to avoid confusion, and force the user to specify something?

Reply:
I think the bigger question is - where should the mirror and gateway live? There are some downsides to having it live in the `linkerd` namespace but it feels like there might be more by having it live by itself. Any ideas what the tradeoffs are? Naively, it feels like everything should go into `linkerd` namespace and not be configurable, but I'm sure I'm missing something.